drivers/sensor/zephyr_thermistor/zephyr_thermistor_device.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_thermistor
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/zephyr_thermistor.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ZEPHYR_THERMISTOR, CONFIG_SENSOR_LOG_LEVEL);
+
+struct zephyr_thermistor_data {
+	struct k_mutex mutex;
+	int16_t raw;
+	int16_t sample_val;
+};
+
+struct zephyr_thermistor_config {
+	const struct adc_dt_spec adc_channel;
+	const struct ntc_config ntc_cfg;
+};
+
+static int zephyr_thermistor_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct zephyr_thermistor_data *data = dev->data;
+	const struct zephyr_thermistor_config *cfg = dev->config;
+	int32_t val_mv;
+	int res;
+	struct adc_sequence sequence = {
+		.options = NULL,
+		.buffer = &data->raw,
+		.buffer_size = sizeof(data->raw),
+		.calibrate = false,
+	};
+
+	k_mutex_lock(&data->mutex, K_FOREVER);
+
+	adc_sequence_init_dt(&cfg->adc_channel, &sequence);
+	res = adc_read(cfg->adc_channel.dev, &sequence);
+	val_mv = data->raw;
+	res = adc_raw_to_millivolts_dt(&cfg->adc_channel, &val_mv);
+	data->sample_val = val_mv;
+
+	k_mutex_unlock(&data->mutex);
+
+	return res;
+}
+
+static int zephyr_thermistor_channel_get(const struct device *dev, enum sensor_channel chan,
+					 struct sensor_value *val)
+{
+	struct zephyr_thermistor_data *data = dev->data;
+	const struct zephyr_thermistor_config *cfg = dev->config;
+	uint32_t ohm, max_adc;
+	int32_t temp;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		max_adc = (1 << (cfg->adc_channel.resolution - 1)) - 1;
+		ohm = ntc_get_ohm_of_thermistor(&cfg->ntc_cfg, max_adc, data->raw);
+		temp = ntc_get_temp_mc(cfg->ntc_cfg.type, ohm);
+		val->val1 = temp / 1000;
+		val->val2 = (temp % 1000) * 1000;
+		LOG_INF("ntc temp says %u", val->val1);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static const struct sensor_driver_api zephyr_thermistor_driver_api = {
+	.sample_fetch = zephyr_thermistor_sample_fetch,
+	.channel_get = zephyr_thermistor_channel_get,
+};
+
+int zephyr_thermistor_reset(const struct device *dev)
```

Comment:
`zephyr_thermistor_reset` is only used during in `zephyr_thermistor_init`, move the implementation there and remove this function.